python/cudf/cudf/tests/utils.py

```diff
@@ -177,8 +183,12 @@ def assert_exceptions_equal(
         raise
     except Exception as e:
         with pytest.raises(
-            type(e),
-            match=re.escape(str(e)) if compare_error_message is True else None,
+            type(e) if expected_exception is None else expected_exception,
+            match=None
+            if not compare_error_message
+            else re.escape(str(e))
+            if expected_error_message is None
+            else expected_error_message,
```

Comment:
Can we wrap this in parenthesis? It makes it very hard to read currently.

Reply:
Done 👍